go-controller/pkg/ovn/controller/services/services_controller_test.go

```diff
@@ -628,3 +632,11 @@ func nodeConfig(nodeName string, nodeIP string) *nodeInfo {
 		switchName:        nodeSwitchName(nodeName),
 	}
 }
+
+func temporarilyEnableGomegaMaxLengthFormat() func() {
+	initialMaxLength := format.MaxLength
+	format.MaxLength = 0
+	return func() {
+		format.MaxLength = initialMaxLength
+	}
+}
```

Comment:
> One thing I had done in the past is having a `init()` function in `pkg/von/testing/libovsdb/matchers.go` to set this. It would be set every time the matcher was imported and that kinda made sense to me. But I guess not to everybody because it was removed :/

I didn't do that because I guess if it doesn't exist, people didn't want it. Which is basically the same you just said.

Thus, I've added it to the only place I needed it, hoping it wouldn't bother anyone else.

> 
> Anyway, the way you do it here is not very canonical for a test. Any reason you don't want to use the tear up/tear down mechanism of the testing framework. ginkgo has BeforeEach, etc....

I agree, but, whoever wrote this test, clearly didn't like ginkgo - it's not even imported, nor used.

Long story short, I don't want to re-write this test to work with ginkgo. I'd rather just drop the func.

Reply:
Ah, ginkgo was just an example. I checked the testing package that is used here and it has a cleanup function, but nothing very sophisticated. 

https://pkg.go.dev/testing#F.Cleanup

This is a bit of a nit but having something more similar to 
```
func setupTest(tb testing.TB) func(tb testing.TB) {...}
```
would probably look more familiar.

Reply:
I feel that belongs on a test framework.

I am simply trying to create a button that enables the full output of the diff to be printed on error - without pushing that to everyone else.

I will happily comply w/ using the `t.Cleanup` func though.